Update Register 211 to be configurable except reserved bit ;)

rysiulg opened the pull request:
I've added also in esp_home project name and version for easly identification running code

rysiulg commented:
I've prepared some commits -thats why so fast. minus after some error in commit -than I must adopt in next commits offline files ;)
next planned:
4.1.5 add COP
4.1.6 update 272 lambda section -to avoid out of range array
4.1.7 update 200
4.1.8 update 101 -status states from heat pump as text sensor
4.1.9 update 7-8-9 as switch -made changes remotelly but is reverted soon, also from HA. maybe i don't have any heater connected and this is the reason ;) -but these are configurable

Mosibi commented:
I already expected some extra pull requests, that’s why I did not created a release yet. Since this PR and the next ones fix some issues and also introduce new “sensors”, I will wait until all your PRs are in and then create a release. That will be 4.2.0 or maybe a 5.0.0 if major changes are being made. But I will decide later on this

Mosibi commented:
> I've added also in esp_home project name and version for easly identification running code

I like it, but also am afraid that updating this will be forgotten. Can you leave out so I can think about an automated manner to do this.

rysiulg commented:
> > I've added also in esp_home project name and version for easly identification running code
> 
> I like it, but also am afraid that updating this will be forgotten. Can you leave out so I can think about an automated manner to do this.

Yes -I've removed version from esp_home. I know it's easy to forgot, but I'm affraid it will be difficult to autoversioning this ;) -It will be grat solution -If you find way give me hint ;)
I think project name can be set -it's nicly displayed in Homeassistant as vwendor ;)

rysiulg commented:
> I already expected some extra pull requests, that’s why I did not created a release yet. Since this PR and the next ones fix some issues and also introduce new “sensors”, I will wait until all your PRs are in and then create a release. That will be 4.2.0 or maybe a 5.0.0 if major changes are being made. But I will decide later on this

OK. I don't change ChangeLog from now ;)

rysiulg commented:
Or only write temprary -what is added changed ;)

Mosibi commented:
> I think project name can be set -it's nicly displayed in Homeassistant as vwendor ;)

Agree, let's set that indeed. For the version I am thinking about a github action on my repo, but I have to dive into that.

Mosibi commented:
I will test this PR today in the evening.

rysiulg commented:
OK. So to monday ;) -I'll go in travel on week ;)
I think everything will be ok. These Register 211 also affects to turn Off all Power on Register 0 ;) -this is service parameters changed heat pump control. Thats why i implemented for me autorecovery power on afetr some time when heatpump was disabled ;)

Mosibi commented:
Do you have information about which register from 0 will be set to off when something from 211 is changed?

Enjoy your trip!

rysiulg commented:
hiunfortunately all1. power on climate for fan coil2. power on zone 1 heat3. power on dhw4. power on zone2 heatfor sure after change value from register 210, 211 and other values like dt5_on delta for dhw and i think all other  delta temp values, also changing emission type (radians, underfloor, fan coil unit) and otherslimits of temp also depends from emission type -eg when i’m on cool mode i power on cool/heat zone 1 and i can set to cool water in buffer to 5 deg if i choose for zone1 cool “fan coil unit” -in other cases temp T1s is limited to 19degPozdrawiamRyszard GwizdakMARM.pl Sp. z o.o.Wiadomość napisana przez Richard Arends ***@***.***> w dniu 19.10.2024, o godz. 15:18:﻿
Do you have information about which register from 0 will be set to off when something from 211 is changed?
Enjoy your trip!

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

Mosibi commented:
> hi unfortunately all

I was already “afraid” of that.

> i can provide solution in eg. 5minut interval to power on heat zone1 and dhw or to get from global to turn on these which was running 

That’s logic I want to keep out of this config. But heaving this knowledge (if you turn this off, then the effect is…) written down somewhere, would nice. Getting that from the manuals is almost un-doable for a normal person 😌

> weak point is reset vs memory write worn if too often these will be saved to rom;)

Indeed, let’s stay away from that.

rysiulg commented:
maybe for essential registers add line disabled_by_default which makes registers not recorded by home assistant at default but these registers must be checked as enabled in home assistant to be able to config ;) -it could be solution for that fragile registers.;)PozdrawiamRyszard GwizdakMARM.pl Sp. z o.o.Wiadomość napisana przez Richard Arends ***@***.***> w dniu 19.10.2024, o godz. 16:59:﻿

hi unfortunately all

I was already “afraid” of that.

i can provide solution in eg. 5minut interval to power on heat zone1 and dhw or to get from global to turn on these which was running

That’s logic I want to keep out of this config. But heaving this knowledge (if you turn this off, then the effect is…) written down somewhere, would nice. Getting that from the manuals is almost un-doable for a normal person 😌

weak point is reset vs memory write worn if too often these will be saved to rom;)

Indeed, let’s stay away from that.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

Mosibi commented on heatpump.yaml:
```diff
@@ -36,6 +40,8 @@ esphome:
   name: "${devicename}"
   comment: "${description}"
   friendly_name: "${description}"
+  project:
```
I got an error on this one: `'version' is a required option for [project].` Sad, but let's remove it for now until I have good option to change the version automatically

Mosibi commented on heatpump.yaml:
```diff
@@ -2821,12 +2702,500 @@ switch:
+  - platform: template
+    name: "Parameter Setting 2 Port Definition"
+    id: "${devicename}_parameter_setting_2_port_definition"
+    disabled_by_default: true
```
Let's not disable this one. It is up to the user if he/she wants to set in a dashboard, but imho we should enable these so that (historic) metrics are always available.

rysiulg replied:
ok
this don’t disable completely. it makes not default visible values in home assistant dashboard, but in home assistant in esp home sensor edit it is possible to enable it from that time is normally recorded sensor in home assistant;)

Mosibi commented on heatpump.yaml:
```diff
@@ -2821,12 +2702,500 @@ switch:
+  - platform: template
+    name: "Parameter Setting 2 Solar Energy Kit Enable"
+    id: "${devicename}_parameter_setting_2_solar_energy_kit_enable"
+    disabled_by_default: true
```
See other note about `disabled_by_default`

Mosibi commented on heatpump.yaml:
```diff
@@ -2821,12 +2702,500 @@ switch:
+  - platform: template
+    name: "Parameter Setting 2 Solar Energy Input Port"
+    id: "${devicename}_parameter_setting_2_solar_energy_input_port"
+    disabled_by_default: true
```
See other note about `disabled_by_default`

Mosibi commented on heatpump.yaml:
```diff
@@ -2821,12 +2702,500 @@ switch:
+  - platform: template
+    name: "Parameter Setting 2 Piping Length Selection"
+    id: "${devicename}_parameter_setting_2_piping_length_selection"
+    disabled_by_default: true
```
See other note about `disabled_by_default`

rysiulg commented:
i corrected this from phone so i hope all i found -on pc you have ctrl+f and is possible to find and replace -on phone is worse…;)

Mosibi commented:
Code is good. I will merge it and immediately change the CHANGELOG.md to reflect the right versioning. Can you put your next changes in the CHANGELOG.md under the `## [Unreleased]` line.
